File: openapi-generator/dart/lib/model/document_field.dart
part of openapi.api;

class DocumentField {
  /* Value of this field if this document field has valueType: ValueType_Number. */
  double numberValue = null;
  /* Value of this field if this document field has valueType: ValueType_Photo. Array of photo objects where each object contains a URL for a photo. */
  List<DocumentFieldCreatePhotoValue> photoValue = [];
  /* Value of this field if this document field has valueType: ValueType_String. */
  String stringValue = null;
  /* Determines the type of this field and what type of value this field has. It should be either ValueType_Number, ValueType_String, or ValueType_Photo. */
  String valueType = null;
  /* Descriptive name of this field. */
  String label = null;
  /* DEPRECATED: Please use stringValue, numberValue, or photoValue instead. Value of this field. Depending on what kind of field it is, this may be one of the following: an array of image urls, a float, an integer, or a string. */
  Object value = null;
  DocumentField();

  @override
  String toString() {
    return 'DocumentField[numberValue=$numberValue, photoValue=$photoValue, stringValue=$stringValue, valueType=$valueType, label=$label, value=$value, ]';
  }

  DocumentField.fromJson(Map<String, dynamic> json) {
    if (json == null) return;
    if (json['numberValue'] == null) {
      numberValue = null;
    } else {
          numberValue = json['numberValue'];
    }
    if (json['photoValue'] == null) {
      photoValue = null;
    } else {
      photoValue = DocumentFieldCreatePhotoValue.listFromJson(json['photoValue']);
    }
    if (json['stringValue'] == null) {
      stringValue = null;
    } else {
          stringValue = json['stringValue'];
    }
    if (json['valueType'] == null) {
      valueType = null;
    } else {
          valueType = json['valueType'];
    }
    if (json['label'] == null) {
      label = null;
    } else {
          label = json['label'];
    }
    if (json['value'] == null) {
      value = null;
    } else {
          value = json['value'];
    }
  }

  Map<String, dynamic> toJson() {
    return {
      'numberValue': numberValue,
      'photoValue': photoValue,
      'stringValue': stringValue,
      'valueType': valueType,
      'label': label,
      'value': value
    };
  }

  static List<DocumentField> listFromJson(List<dynamic> json) {
    return json == null ? new List<DocumentField>() : json.map((value) => new DocumentField.fromJson(value)).toList();
  }

  static Map<String, DocumentField> mapFromJson(Map<String, dynamic> json) {
    var map = new Map<String, DocumentField>();
    if (json != null && json.length > 0) {
      json.forEach((String key, dynamic value) => map[key] = new DocumentField.fromJson(value));
    }
    return map;
  }
}

File: openapi-generator/csharp-dotnet2/src/main/CsharpDotNet2/Org/OpenAPITools/Model/DocumentField.cs
using System;
using System.Text;
using System.Collections;
using System.Collections.Generic;
using System.Runtime.Serialization;
using Newtonsoft.Json;

namespace Org.OpenAPITools.Model {

  /// <summary>
  /// 
  /// </summary>
  [DataContract]
  public class DocumentField {
    /// <summary>
    /// Value of this field if this document field has valueType: ValueType_Number.
    /// </summary>
    /// <value>Value of this field if this document field has valueType: ValueType_Number.</value>
    [DataMember(Name="numberValue", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "numberValue")]
    public double? NumberValue { get; set; }

    /// <summary>
    /// Value of this field if this document field has valueType: ValueType_Photo. Array of photo objects where each object contains a URL for a photo.
    /// </summary>
    /// <value>Value of this field if this document field has valueType: ValueType_Photo. Array of photo objects where each object contains a URL for a photo.</value>
    [DataMember(Name="photoValue", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "photoValue")]
    public List<DocumentFieldCreatePhotoValue> PhotoValue { get; set; }

    /// <summary>
    /// Value of this field if this document field has valueType: ValueType_String.
    /// </summary>
    /// <value>Value of this field if this document field has valueType: ValueType_String.</value>
    [DataMember(Name="stringValue", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "stringValue")]
    public string StringValue { get; set; }

    /// <summary>
    /// Determines the type of this field and what type of value this field has. It should be either ValueType_Number, ValueType_String, or ValueType_Photo.
    /// </summary>
    /// <value>Determines the type of this field and what type of value this field has. It should be either ValueType_Number, ValueType_String, or ValueType_Photo.</value>
    [DataMember(Name="valueType", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "valueType")]
    public string ValueType { get; set; }

    /// <summary>
    /// Descriptive name of this field.
    /// </summary>
    /// <value>Descriptive name of this field.</value>
    [DataMember(Name="label", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "label")]
    public string Label { get; set; }

    /// <summary>
    /// DEPRECATED: Please use stringValue, numberValue, or photoValue instead. Value of this field. Depending on what kind of field it is, this may be one of the following: an array of image urls, a float, an integer, or a string.
    /// </summary>
    /// <value>DEPRECATED: Please use stringValue, numberValue, or photoValue instead. Value of this field. Depending on what kind of field it is, this may be one of the following: an array of image urls, a float, an integer, or a string.</value>
    [DataMember(Name="value", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "value")]
    public Object Value { get; set; }


    /// <summary>
    /// Get the string presentation of the object
    /// </summary>
    /// <returns>String presentation of the object</returns>
    public override string ToString()  {
      var sb = new StringBuilder();
      sb.Append("class DocumentField {\n");
      sb.Append("  NumberValue: ").Append(NumberValue).Append("\n");
      sb.Append("  PhotoValue: ").Append(PhotoValue).Append("\n");
      sb.Append("  StringValue: ").Append(StringValue).Append("\n");
      sb.Append("  ValueType: ").Append(ValueType).Append("\n");
      sb.Append("  Label: ").Append(Label).Append("\n");
      sb.Append("  Value: ").Append(Value).Append("\n");
      sb.Append("}\n");
      return sb.ToString();
    }

    /// <summary>
    /// Get the JSON string presentation of the object
    /// </summary>
    /// <returns>JSON string presentation of the object</returns>
    public string ToJson() {
      return JsonConvert.SerializeObject(this, Formatting.Indented);
    }

}
}
